### PR TITLE
Statsdcc <-> statsd compatibility improvements

### PR DESCRIPTION
**Pull request by rrusso1982**
Statsdcc is a multi threaded replacement for statsd that is written in c++.  It is being tested as a viable replacement for our statsd installation to improve ingestion.  Unfortunately statsdcc is not 100% compatible with statsd making is not a "drop-in" replacement for the product.  The effort of this PR is to improve the compatibility between statsd and statsdcc.

The implemented improvement are:

- Metrics prefix: The ability to add a prefix to all metrics that are processed by the statsd instance.  Ex: counters.concierge.X.count -> production.counters.concierge.X.count
- Metrics grouping: All metrics that go through statsdcc should be grouped by metrics type.  These are counters, timers, gauges, and sets. Hence instead of producing the counter "concierge.X.count" it will produce "counters.concierge.X.count"
- Timer percentiles: Statsdcc does support percentile calculations for timers however it does not produce the same set that statsd produces.  By default it will produce upper_X for each percentile but based on graph samplings in grafana we require count_X, sum_X, and mean_X.

### CODE REVIEW

File: src/ledger.cc [hunk 1]
```diff
@@ -168,23 +168,34 @@ void Ledger::process() {
           ++threshold_itr) {
         double pct =
           (*threshold_itr < 0) ? -(*threshold_itr) : (*threshold_itr);
+         int num_in_threshold = 0;
```
Comment (Jonnymcc) [inline src/ledger.cc]: Why make `num_in_threshold` an `int` instead of `double`?

Comment (rrusso1982) [reply src/ledger.cc]: It's used as an array pointer so it has to be an int to hit an index in the array.

Comment (vasiliyb) [reply src/ledger.cc]: unsigned long

File: src/ledger.cc [hunk 2]
```diff
@@ -168,23 +168,34 @@ void Ledger::process() {
...
+        double pct_sum = 0;
+        for (int i = 0; i < num_in_threshold; ++i) {
+            pct_sum += values[i];
+        }
+        double pct_mean = pct_sum / num_in_threshold;
+
         // generate pct name
         char clean_pct[17] = {0};
-        snprintf(clean_pct, sizeof(clean_pct), "upper_%g", pct);
+        snprintf(clean_pct, sizeof(clean_pct), "%g", pct);
```
Comment (vasiliyb) [inline src/ledger.cc:190]: nit: can comment

**Review by vasiliyb — APPROVED** [review]
lgtme, maybe add a comment that we always have a prefix

File: src/ledger.cc (hunk 2 above)
Comment (rrusso1982) [reply src/ledger.cc:190]: I recall what this was for.  This takes the value of pct and injects it into the variable clean_pct with formatting before it get converted to a display value.  So this is necessary I think.

**Review by zihaoyu — APPROVED** [review]
lgtm

**Pull request merged**